### PR TITLE
Fixes #10981 - Migrate logging to new custom logging in core

### PR DESCRIPTION
**Pull request by adamruzicka**

### CODE REVIEW

**Comment by domcleal** [comment]
I'd recommend also shipping an example config (YAML) file with the loggers config commented out, so people can easily see the defaults and change it.  This also needs a change to requires_foreman to `>= 1.9.0`.

File: app/models/foreman_tasks/task/dynflow_task.rb [hunk 1]
```diff
@@ -85,15 +85,16 @@ def get_humanized(method)
...
         begin
           changes = task.update_from_dynflow(task.execution_plan.to_hash)
           unless changes.empty?
             fixed_count += 1
-            Rails.logger.warn("Task %s updated at consistency check: %s" % [task.id, changes.inspect])
+            logger.warn("Task %s updated at consistency check: %s" % [task.id, changes.inspect])
           end
         rescue => e
-          Rails.logger.warn("Failed at consistency check for task %s: %s\n %s" % [task.id, e.message, e.backtrace.join("\n")])
+          logger.warn("Failed at consistency check for task %s: %s\n %s" % [task.id, e.message, e.backtrace.join("\n")])
```
Comment (iNecas) [inline app/models/foreman_tasks/task/dynflow_task.rb]: Could you also search for this kind of exception logging and convert that to use https://github.com/theforeman/foreman/blob/develop/lib/foreman/logging.rb#L60 ?

File: lib/foreman_tasks/engine.rb [hunk 2]
```diff
@@ -4,6 +4,11 @@ class Engine < ::Rails::Engine
 
     initializer 'foreman_tasks.load_default_settings', :before => :load_config_initializers do
       require_dependency File.expand_path('../../../app/models/setting/foreman_tasks.rb', __FILE__) if (Setting.table_exists? rescue(false))
+      logging = Foreman::Plugin::Logging.new('foreman-tasks')
+      logging.configure(:loggers => {
```
Comment (domcleal) [inline lib/foreman_tasks/engine.rb]: Use the plugin API if you can: http://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Plugin#Custom-Plugin-Loggers

File: lib/foreman_tasks/engine.rb [hunk 3]
```diff
@@ -4,6 +4,11 @@ class Engine < ::Rails::Engine
 
     initializer 'foreman_tasks.load_default_settings', :before => :load_config_initializers do
       require_dependency File.expand_path('../../../app/models/setting/foreman_tasks.rb', __FILE__) if (Setting.table_exists? rescue(false))
+      logging = Foreman::Plugin::Logging.new('foreman-tasks')
```
Comment (domcleal) [inline lib/foreman_tasks/engine.rb]: That's already created by default.

File: lib/foreman_tasks/dynflow/daemon.rb [hunk 4]
```diff
@@ -47,10 +47,11 @@ def run_background(command = "start", options = {})
                        :log_output => true,
                        :ARGV => [command]) do |*args|
         begin
+          ::Logging.reopen
           run(options[:foreman_root])
         rescue => e
           STDERR.puts e.message
-          Rails.logger.fatal e
+          Foreman::Logging.exception("", e)
```
Comment (iNecas) [inline lib/foreman_tasks/dynflow/daemon.rb]: "Failed running foreman-tasks daemon"

**Comment by iNecas** [comment]
ACK after fixing the last comment

**Comment by iNecas** [comment]
Thanks!

**Pull request merged**